### PR TITLE
[LMS-344][FE][Markup] Sign-in Page

### PR DESCRIPTION
**Pull request by mikorosales**
## Issue Link
[LMS-344](https://framgiaph.backlog.com/view/LMS-344)

## Defintion of Done
- [x] Sign-in page /auth/sign-in
- [x] Must have two inputs (email and password)
- [x] Must implement frontend validation
- [x] Must not submit if it has FE validation
- [x] Must show an alert for the FE validation

## Steps to reproduce
http://localhost:3000/auth/sign-in

## Affected Components / Functionalities / Page
NavBar

## Test Cases
To test localStorage conditional rendering

1. Go to the NavBar component index
     - src > shared > components > Navbar
3. In line 28, set value to true

## Notes
I used localStorage for conditional rendering

## Screenshots
signin
image

### CODE REVIEW

**Comment by Marielleregner7** [comment]
Comment:

- [ ] Please fix build error 🙇
image

**Review by Marielleregner7 — APPROVED** [review]
**APPROVED**
- [x] Must have two inputs (email and password)
- [ ]  Must implement frontend validation
- [x] Must not submit if it has FE validation
- [x] Must show an alert for the FE validation
- [x] No ESLint warnings or errors.

Comments: 0
Test time: 110 mins
Iterations: 0

Proof:
signinmarkupmiko
image

File: client/package-lock.json
Comment (kdacudag) [inline client/package-lock.json:1]: Remove

File: node_modules/.package-lock.json
Comment (kdacudag) [inline node_modules/.package-lock.json:1]: Remove

File: package-lock.json
Comment (kdacudag) [inline package-lock.json:1]: Remove

File: client/src/shared/components/Navbar/index.tsx [hunk 1]
```diff
@@ -18,9 +17,20 @@ export interface NavbarProps {
   isSticky?: boolean;
```
Comment (ragg09-sunasterisk) [inline client/src/shared/components/Navbar/index.tsx:17]: dont pass this as prop, navbar is always fixed at top

File: client/src/shared/components/Navbar/index.tsx [hunk 2]
```diff
@@ -18,9 +17,20 @@ export interface NavbarProps {
   isSticky?: boolean;
 }
 
-const Navbar: React.FC<NavbarProps> = ({ navItems, dropdownItems, isSticky }) => {
+const Navbar: React.FC<NavbarProps> = ({
+  navItems,
+  dropdownItems,
+  isSticky
+}) => {
+  // this is the flag for the conditional rendering
```
Comment (ragg09-sunasterisk) [inline client/src/shared/components/Navbar/index.tsx:25]: remove comment

File: client/src/shared/components/Navbar/index.tsx [hunk 3]
```diff
@@ -18,9 +17,20 @@ export interface NavbarProps {
...
 }
 
-const Navbar: React.FC<NavbarProps> = ({ navItems, dropdownItems, isSticky }) => {
+const Navbar: React.FC<NavbarProps> = ({
+  navItems,
+  dropdownItems,
+  isSticky
+}) => {
+  // this is the flag for the conditional rendering
+  const signedIn =
```
Comment (ragg09-sunasterisk) [inline client/src/shared/components/Navbar/index.tsx:26]: create a utils out of this, we can reuse this in future
you can name it 
eg.
```
export const isLoggedIn = (): boolean => {
 . . .
};
```

**Review by kdacudag — APPROVED** [review]
Approved

File: client/src/shared/utils/isSignedIn.ts
Comment (ragg09-sunasterisk) [inline client/src/shared/utils/isSignedIn.ts:1]: - please rename this as auth.ts
- export it using the utils/index.ts

import should look like this 
`import { isSignedIn } from '../../utils';`

File: client/src/shared/components/Navbar/index.tsx [hunk 4]
```diff
@@ -34,37 +39,49 @@ const Navbar: React.FC<NavbarProps> = ({ navItems, dropdownItems, isSticky }) =>
...
             </div>
           </div>
-          <div className="flex items-center">
-            <Link href={'/settings'} className="text-gray-600">
-              <SettingsIcon width={20} height={20} className="mr-2" />
-            </Link>
-            <Avatar name="Jane Doe" />
-            <Dropdown options={dropdownItems} label="John Doe" />
-          </div>
+          {signedIn
```
Comment (ragg09-sunasterisk) [inline client/src/shared/components/Navbar/index.tsx:61]: no need to use useStatae you can use directly the function

```suggestion
          {isSignedIn() ? (
```

**Review by ragg09-sunasterisk — APPROVED** [review]

**Comment by ragg09-sunasterisk** [comment]
Comments: 8
Iterations: 3

**Pull request merged**